components/PoeModal.tsx

```diff
@@ -0,0 +1,537 @@
+    undefined
+  );
+  const [timeoutAndHash, setTimeoutAndHash] = useState([false, ""]);
+  const [currentState, setCurrentState] = useState(State.IDLE);
+
+  const version =
+    state.contracts[state.currentContract ?? ""]?.version ??
+    state.currentStorage?.version;
+
+  const rows = useMemo(
+    () =>
+      (transfers ?? []).map(t =>
+        contentToData(version, transferToProposalContent(t), walletTokens ?? [])
+      ),
+    [transfers]
+  );
+  useEffect(() => {
+    if (!state.p2pClient) return;
+
+    const challengeCb = setMessage;
+    const transactionCb = async (message: OperationRequestOutput) => {
+      setCurrentMetadata([message.id, message.appMetadata]);
+
+      setTransfers(
+        (
+          await Promise.all(
+            message.operationDetails.map(async detail => {
+              switch (detail.kind) {
+                case TezosOperationType.TRANSACTION:
+                  if (!!detail.parameters) {
+                    try {
+                      const contract = await state.connection.contract.at(
+                        detail.destination
+                      );
+
+                      const methodSchema = new Schema(
+                        contract.entrypoints.entrypoints[
+                          detail.parameters.entrypoint
+                        ]
+                      );
+
+                      const value = contract.methods[
+                        detail.parameters.entrypoint
+                      ](
+                        methodSchema.Execute(detail.parameters.value)
+                      ).toTransferParams().parameter?.value;
+
+                      if (!value) return undefined;
+                      const param = emitMicheline(value as Expr);
+
+                      const parser = new Parser();
+
+                      const type = emitMicheline(
+                        parser.parseJSON(
+                          contract.entrypoints.entrypoints[
+                            detail.parameters.entrypoint
+                          ]
+                        ),
+                        {
+                          indent: "",
+                          newline: "",
+                        }
+                      );
+
+                      return {
+                        type: "contract",
+                        values: {
+                          lambda: generateExecuteContractMichelson(version, {
+                            address: detail.destination,
+                            amount: Number(detail.amount),
+                            entrypoint: detail.parameters.entrypoint,
+                            type,
+                            param,
+                          }),
+                          metadata: null,
+                        },
+                      };
+                    } catch (e) {
+                      console.log("Error while converting contract call", e);
```

Comment:
- the error message should in the UI.
- also need to handle the error by sending/rejecting response back to client
![image](https://github.com/marigold-dev/tzsafe-ui/assets/115862/fc7add93-1b48-4b10-838d-d29487f7635d)

Reply:
another suggestion:
the pop window should also show which tzsafe wallet is going to created a proposal.
If users import multiple and connected isn't the current one, it will confusing the users.